src/ClientData/CallstackDataTest.cpp

```diff
@@ -200,4 +200,144 @@ TEST(CallstackData, FilterCallstackEventsBasedOnMajorityStartExcludesFunctionToS
                                                                          event4, event5, event6}));
 }
 
+TEST(CallstackData, ForEachCallstackEventOfTidInTimeRangeDiscretized) {
+  CallstackData callstack_data;
+
+  constexpr uint32_t kTid = 42;
+  constexpr uint32_t kAnotherTid = 43;
+
+  constexpr uint64_t kCloneAddress = 0x10;
+
+  const uint64_t cs1_id = 12;
+  const uint64_t cs1_outer = kCloneAddress;
+  const uint64_t cs1_inner = 0x11;
+  CallstackInfo cs1{{cs1_inner, cs1_outer}, CallstackType::kComplete};
+  callstack_data.AddUniqueCallstack(cs1_id, std::move(cs1));
+
+  const uint64_t cs2_id = 13;
+  const uint64_t cs2_outer = kCloneAddress;
+  const uint64_t cs2_inner = 0x21;
+  CallstackInfo cs2{{cs2_inner, cs2_outer}, CallstackType::kComplete};
```

Comment:
Do you even need two distinct `CallstackInfo`s for this test? I feel like a single `CallstackInfo` with no frames at all would serve the same purpose.

Reply:
In that case the other two comments I left on this test are irrelevant.